app/helpers/patient_helper.rb

```diff
@@ -76,4 +76,71 @@ def normalize_name(name)
 
     name.delete(" \t\r\n").downcase
   end
+
+  def timezone_for_state(name)
+    timezones = {
+      'alabama' => '-05:00',
+      'alaska' => '-08:00',
+      'americansamoa' => '-11:00',
+      'arizona' => '-07:00',
+      'arkansas' => '-05:00',
+      'california' => '-07:00',
+      'colorado' => '-06:00',
+      'connecticut' => '-04:00',
+      'delaware' => '-04:00',
+      'districtofcolombia' => '-04:00',
+      'federatedstatesofmicronesia' => '+11:00',
+      'florida' => '-04:00',
+      'georgia' => '-04:00',
+      'guam' => '+10:00',
+      'hawaii' => '-10:00',
+      'idaho' => '-06:00',
+      'illinois' => '-05:00',
+      'indiana' => '-04:00',
+      'iowa' => '-05:00',
+      'kansas' => '-05:00',
+      'kentucky' => '-04:00',
+      'louisiana' => '-05:00',
+      'maine' => '-04:00',
+      'marshallislands' => '+12:00',
+      'maryland' => '-04:00',
+      'massachusetts' => '-04:00',
+      'michigan' => '-04:00',
+      'minnesota' => '-05:00',
+      'mississippi' => '-05:00',
+      'missouri' => '-05:00',
+      'montana' => '-06:00',
+      'nebraska' => '-05:00',
+      'nevada' => '-07:00',
+      'newhampshire' => '-04:00',
+      'newjersey' => '-04:00',
+      'newmexico' => '-06:00',
+      'newyork' => '-04:00',
+      'northcarolina' => '-04:00',
+      'northdakota' => '-05:00',
+      'northernmarianaislands' => '+10:00',
+      'ohio' => '-04:00',
+      'oklahoma' => '-05:00',
+      'oregon' => '-07:00',
+      'palau' => '+09:00',
+      'pennsylvania' => '-04:00',
+      'puertorico' => '-04:00',
+      'rhodeisland' => '-04:00',
+      'southcarolina' => '-04:00',
+      'southdakota' => '-05:00',
+      'tennessee' => '-05:00',
+      'texas' => '-05:00',
+      'utah' => '-06:00',
+      'vermont' => '-04:00',
+      'virginislands' => '-04:00',
+      'virginia' => '-04:00',
+      'washington' => '-07:00',
+      'westvirginia' => '-04:00',
+      'wisconsin' => '-05:00',
+      'wyoming' => '-06:00',
+      nil => '-04:00',
+      '' => '-04:00'
+    }
+    timezones[normalize_name(name)] || '-04:00'
```

Comment:
I see nil covered above, but a nil would be passed to `normalize_name(name)` first right? So it might do a `name.delete` on a nil, which would halt the entire notification job.

Reply:
it wouldn't get to the name.delete line, there's already a guard. https://github.com/SaraAlert/SaraAlert/blob/a0ccb56c8f85de88d6ce36a1b60fe7a1d67c33fc/app/helpers/patient_helper.rb#L75

Reply:
Oh nice, didn't see that in the preview.